### DISCUSSION TITLE
I would like to revert GitHub comments to previous versions

### DISCUSSION THREAD

**Discussion by Neykah** [community/community#52479] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hi,

I'm glad that GitHub issues allow us to see the whole comments history in the issue thread and browse the diffs efficiently.
However, it seems impossible to revert a comment to a previous version.
When clicking on a previous comment version, the diff is displayed in rich format, and I cannot get the original markdown of that version.
It's possible to copy-paste the diff content, remove the unused parts, and re-apply formatting manually, but it's very cumbersome.
Would it be possible to get a simple revert feature or display the raw markdown content of a GitHub comment at any time in history?

**Comment by szszszsz** [comment]
Hey! Seconding on that. Just to add for a possible temporary/passable solution - even adding a hidden HTML element with the previous markdown comment version would do. Nothing to do in the UI. It does not need to be pretty (at least at first).
Edit: or just allow to get these through GH CLI.

**Comment by beedell-roke** [comment]
Just had https://github.com/Stremio/stremio-service/issues/80 revert itself to an older version. I need to be able to restore so I don't lose work, so please add this. Luckily, I had copied the source before I submitted it. Look at its edit history.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/community/discussions/52479#discussioncomment-10763931

Luckily, I had copied the source before I submitted it. Look at its edit history.

**Comment by Z3R0-cmd** [comment]
I just ran into some other issue where clicking a checkbox in another chrome tab after clicking it in the first tab causes the content of the discussion post to be cleared. I'd like to be able to revert it, but there doesn't seem to be a way to do that yet.

Here is the temporary solution I used:

1. Open the history and select the previous version you want to extract the content of
2. Use inspect element to copy the HTML source of the `<article>` tag containing the previous version
3. Save that HTML to a file and then use this python script to convert it to markdown

```py
#!/usr/bin/env python3

import re, sys
from bs4 import BeautifulSoup, NavigableString, Tag

def normalize_ws(s):
    return re.sub(r"[ \t\r\f\v]+", " ", s).strip()

def text_of(tag):
    return normalize_ws(tag.get_text(" ", strip=True))

def pre_to_fence(pre_tag):
    code_tag = pre_tag.find("code") or pre_tag
    content = code_tag.get_text().rstrip("\n")
    return f"\n```\n{content}\n```\n"

def inline_code_to_md(code_tag):
    return f"`{code_tag.get_text().strip()}`"

def node_to_md(node):
    if isinstance(node, NavigableString):
        return normalize_ws(str(node))
    if not isinstance(node, Tag):
        return ""
    name = node.name.lower()

    if name in ("div", "span"):
        parts = [node_to_md(ch) for ch in node.children]
        parts = [p for p in parts if p]
        return (" ".join(parts)).strip()
    if name.startswith("h") and name[1:].isdigit():
        level = max(1, min(int(name[1:]), 6))
        return f"\n{'#' * level} {text_of(node)}\n"
    if name == "p":
        return f"\n{text_of(node)}\n"
    if name == "br":
        return "\n"
    if name == "code":
        return "" if isinstance(node.parent, Tag) and node.parent.name.lower() == "pre" else inline_code_to_md(node)
    if name == "pre":
        return pre_to_fence(node)
    if name == "ul":
        items = []
        for li in node.find_all("li", recursive=False):
            md = node_to_md(li).strip()
            if md:
                items.append(f"- {md}")
        return "\n" + "\n".join(items) + "\n"
    if name == "li":
        parts = [node_to_md(ch) for ch in node.children]
        return " ".join([p for p in parts if p]).strip()
    parts = [node_to_md(ch) for ch in node.children]
    parts = [p for p in parts if p]
    return (" ".join(parts)).strip() if parts else ""

def html_to_md(html):
    bs = BeautifulSoup(html, "html.parser")
    root = bs.find("div") or bs
    md = "\n".join(p for p in (node_to_md(ch) for ch in root.children) if p).strip()
    md = md.replace("\n\n\n", "\n\n")
    md = "\n".join(line.rstrip() for line in md.splitlines())
    return md.strip() + "\n"

def main():
    if len(sys.argv) != 2:
        print(f"Usage: {sys.argv[0]} input.html")
        sys.exit(2)
    with open(sys.argv[1], "r", encoding="utf-8") as fp:
        sys.stdout.write(html_to_md(fp.read()))

if __name__ == "__main__":
    main()
```

And this is a link to the post: https://github.com/orgs/grimmory-tools/discussions/2056
